make build-and-push-images fails on M1

For me make build-and-push-images fails. Below is an excerpt

```
make build-and-push-images 
cd scripts && ./build-and-push-images.sh
Building image localhost:5000/ods/ods-buildah...
[+] Building 1.6s (18/18) FINISHED                                                                                                                                                                                                        
...
Building image localhost:5000/ods/ods-finish...
...
Building image localhost:5000/ods/ods-go-toolset...
...
Building image localhost:5000/ods/ods-gradle-toolset...
...
Building image localhost:5000/ods/ods-helm...
[+] Building 9.6s (15/21)                                                                                                                                                                                                                 
 => [internal] load build definition from Dockerfile.helm                                                                                                                                                                            0.0s
 => => transferring dockerfile: 2.18kB                                                                                                                                                                                               0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                    0.0s
 => => transferring context: 2B                                                                                                                                                                                                      0.0s
 => [internal] load metadata for registry.access.redhat.com/ubi8/ubi-minimal:8.4                                                                                                                                                     0.5s
 => [internal] load metadata for registry.access.redhat.com/ubi8/go-toolset:1.16.12                                                                                                                                                  0.5s
 => [builder 1/9] FROM registry.access.redhat.com/ubi8/go-toolset:1.16.12@sha256:287d341349331a19866a2862a55e0cc5c412f21a86ba80629f59579ef6a38c03                                                                                    0.0s
 => [internal] load build context                                                                                                                                                                                                    0.0s
 => => transferring context: 419.82kB                                                                                                                                                                                                0.0s
 => [stage-1 1/7] FROM registry.access.redhat.com/ubi8/ubi-minimal:8.4@sha256:c536d4c63253318fdfc1db499f8f4bb0881db7fbd6f3d1554b4d54c812f85cc7                                                                                       0.0s
 => CACHED [stage-1 2/7] RUN microdnf install skopeo-1.4* git-2.27* tar-1.30* && microdnf clean all                                                                                                                                  0.0s
 => CACHED [builder 2/9] RUN mkdir -p /tmp/helm     && cd /tmp     && curl -LO https://get.helm.sh/helm-v3.5.2-linux-amd64.tar.gz     && tar -zxvf helm-v3.5.2-linux-amd64.tar.gz -C /tmp/helm     && mv /tmp/helm/linux-amd64/helm  0.0s
 => CACHED [builder 3/9] COPY go.* .                                                                                                                                                                                                 0.0s
 => CACHED [builder 4/9] COPY cmd cmd                                                                                                                                                                                                0.0s
 => CACHED [builder 5/9] COPY internal internal                                                                                                                                                                                      0.0s
 => CACHED [builder 6/9] COPY pkg pkg                                                                                                                                                                                                0.0s
 => CACHED [builder 7/9] RUN cd cmd/deploy-with-helm && CGO_ENABLED=0 go build -o /usr/local/bin/deploy-with-helm                                                                                                                    0.0s
 => ERROR [builder 8/9] RUN mkdir -p /tmp/sops     && cd /tmp/sops     && curl -LO https://github.com/mozilla/sops/releases/download/v3.7.1/sops-3.7.1-1.x86_64.rpm     && yum install -y sops-3.7.1-1.x86_64.rpm     && sops --ver  9.0s
------                                                                                                                                                                                                                                    
 > [builder 8/9] RUN mkdir -p /tmp/sops     && cd /tmp/sops     && curl -LO https://github.com/mozilla/sops/releases/download/v3.7.1/sops-3.7.1-1.x86_64.rpm     && yum install -y sops-3.7.1-1.x86_64.rpm     && sops --version:         
#15 0.235   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current                                                                                                                                                 
#15 0.235                                  Dload  Upload   Total   Spent    Left  Speed                                                                                                                                                   
100   660  100   660    0     0   2608      0 --:--:-- --:--:-- --:--:--  2608                                                                                                                                                            
100 12.7M  100 12.7M    0     0  6871k      0  0:00:01  0:00:01 --:--:-- 8347k                                                                                                                                                            
#15 2.383 Updating Subscription Management repositories.
#15 2.383 Unable to read consumer identity
#15 2.387 
#15 2.387 This system is not registered with an entitlement server. You can use subscription-manager to register.
#15 2.387 
#15 4.460 Red Hat Universal Base Image 8 (RPMs) - BaseOS  283 kB/s | 570 kB     00:02    
#15 6.616 Red Hat Universal Base Image 8 (RPMs) - AppStre 1.1 MB/s | 2.3 MB     00:02    
#15 8.737 Red Hat Universal Base Image 8 (RPMs) - CodeRea 6.9 kB/s |  13 kB     00:01    
#15 8.897 Error: 
#15 8.897  Problem: conflicting requests
#15 8.897   - package sops-3.7.1-1.x86_64 does not have a compatible architecture
#15 8.898 (try to add '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages)
------
executor failed running [/bin/bash -o pipefail -c mkdir -p /tmp/sops     && cd /tmp/sops     && curl -LO https://github.com/mozilla/sops/releases/download/v${SOPS_VERSION}/sops-${SOPS_VERSION}-1.x86_64.rpm     && yum install -y sops-${SOPS_VERSION}-1.x86_64.rpm     && sops --version]: exit code: 1
make: *** [build-and-push-images] Error 1                                           
```
The above is from the second run. The full log from the first run is at [build-and-push-images.log](https://github.com/opendevstack/ods-pipeline/files/7792330/build-and-push-images.log)

Here is some info about the machine

```
uname -a
Darwin <host-name-redacted> 21.2.0 Darwin Kernel Version 21.2.0: Sun Nov 28 20:28:41 PST 2021; root:xnu-8019.61.5~1/RELEASE_ARM64_T6000 arm64

 docker version
Client:
 Cloud integration: v1.0.22
 Version:           20.10.11
 API version:       1.41
 Go version:        go1.16.10
 Git commit:        dea9396
 Built:             Thu Nov 18 00:36:09 2021
 OS/Arch:           darwin/arm64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.11
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.9
  Git commit:       847da18
  Built:            Thu Nov 18 00:34:44 2021
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.4.12
  GitCommit:        7b11cfaabd73bb80907dd23182b9347b4245eb5d
 runc:
  Version:          1.0.2
  GitCommit:        v1.0.2-0-g52b36a2
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

I believe docker build builds for the os platform by default (arm64) and succeeds for ods-buildah, ods-finish, ods-go-toolset, ods-gradle-toolset but fails for ods-helm

--- a/Makefile
+++ b/Makefile
@@ -104,7 +104,16 @@ install-tekton-pipelines: ## Install Tekton pipelines in KinD cluster.
 .PHONY: install-tekton-pipelines
 
 build-and-push-images: ## Build and push images to local registry.
-		cd scripts && ./build-and-push-images.sh
+		cd scripts && ./build-and-push-images.sh --image start
+		cd scripts && ./build-and-push-images.sh --image finish
+		cd scripts && ./build-and-push-images.sh --image buildah
+		cd scripts && ./build-and-push-images.sh --image sonar
+		cd scripts && ./build-and-push-images.sh --image webhook-interceptor
+		cd scripts && ./build-and-push-images.sh --image go-toolset
+		cd scripts && ./build-and-push-images.sh --image gradle-toolset
+		cd scripts && ./build-and-push-images.sh --image python-toolset
+		cd scripts && ./build-and-push-images.sh --image helm
+		cd scripts && ./build-and-push-images.sh --image node16-typescript-toolset --platform linux/amd64
 .PHONY: build-and-push-images
 
 install-ods-tasks-kind: ## KinD only! Apply ODS ClusterTask manifests in KinD

--- a/scripts/build-and-push-images.sh
+++ b/scripts/build-and-push-images.sh
@@ -27,8 +27,8 @@ while [[ "$#" -gt 0 ]]; do
     -i|--image) IMAGES="$2"; shift;;
     -i=*|--image=*) IMAGES="${1#*=}";;
 
-    -i|--platform) PLATFORM="$2"; shift;;
-    -i=*|--platform=*) PLATFORM="${1#*=}";;
+    -p|--platform) PLATFORM="$2"; shift;;
+    -p=*|--platform=*) PLATFORM="${1#*=}";;
 
     *) echo "Unknown parameter passed: $1"; exit 1;;
 esac; shift; done
@@ -39,6 +39,7 @@ build_and_push_image() {
     odsImage="ods-$image"
     if [ "${SKIP_BUILD}" != "true" ]; then
         echo "Building image $REGISTRY/$NAMESPACE/$odsImage..."
+        # shellcheck disable=SC2086
         docker build $platform_arg \
             --build-arg http_proxy="$http_proxy" \
             --build-arg https_proxy="$https_proxy" \
@@ -50,9 +51,9 @@ build_and_push_image() {
     docker push "$REGISTRY/$NAMESPACE/$odsImage"
 }
 
-platform_arg=""
-if [ ! -z "$PLATFORM" ]; then
-    platform_arg="--platform $PLATFORM"
+platform_arg=
+if [ -n "$PLATFORM" ]; then
+    platform_arg="--platform=${PLATFORM}"
 fi
 
 if [ -z "$IMAGES" ]; then

--- a/scripts/build-and-push-images.sh
+++ b/scripts/build-and-push-images.sh
@@ -14,6 +14,9 @@ IMAGES=""
 http_proxy="${http_proxy:-}"
 https_proxy="${https_proxy:-}"
 
+PLATFORM=""
+# eg. --platform linux/amd64
+
 while [[ "$#" -gt 0 ]]; do
     case $1 in
 
@@ -24,6 +27,9 @@ while [[ "$#" -gt 0 ]]; do
     -i|--image) IMAGES="$2"; shift;;
     -i=*|--image=*) IMAGES="${1#*=}";;
 
+    -i|--platform) PLATFORM="$2"; shift;;
+    -i=*|--platform=*) PLATFORM="${1#*=}";;
+
     *) echo "Unknown parameter passed: $1"; exit 1;;
 esac; shift; done
 
@@ -33,7 +39,7 @@ build_and_push_image() {
     odsImage="ods-$image"
     if [ "${SKIP_BUILD}" != "true" ]; then
         echo "Building image $REGISTRY/$NAMESPACE/$odsImage..."
-        docker build \
+        docker build $platform_arg \
             --build-arg http_proxy="$http_proxy" \
             --build-arg https_proxy="$https_proxy" \
             --build-arg HTTP_PROXY="$http_proxy" \
@@ -43,6 +49,11 @@ build_and_push_image() {
     echo "Pushing image to $REGISTRY/$NAMESPACE/$odsImage ..."
     docker push "$REGISTRY/$NAMESPACE/$odsImage"
 }
+
+platform_arg=""
+if [ ! -z "$PLATFORM" ]; then
+    platform_arg="--platform $PLATFORM"
+fi
 
 if [ -z "$IMAGES" ]; then
     for file in build/package/Dockerfile.*; do

